src/autogluon/bench/cloud/aws/batch_stack/lambdas/lambda_function.py

```diff
@@ -0,0 +1,186 @@
+
+    Sample of cloud_configs.yaml to be supplied by user
+
+    # Infra configurations
+    cdk_context:
+        CDK_DEPLOY_ACCOUNT: dummy
+        CDK_DEPLOY_REGION: dummy
+
+    # Benchmark configurations
+    module: multimodal
+    mode: aws
+    benchmark_name: test_yaml
+    metrics_bucket: autogluon-benchmark-metrics
+
+    # Module specific configurations
+    module_configs:
+        # Multimodal specific
+        multimodal:
+            git_uri#branch: 
+                - https://github.com/autogluon/autogluon#master
+            data_path:
+                - MNIST
+
+        # Tabular specific
+        # You can refer to AMLB (https://github.com/openml/automlbenchmark#quickstart) for more details
+        tabular:
+            framework:
+                - AutoGluon
+            label:
+                - stable
+            amlb_benchmark:
+                - test
+                - small
+            amlb_task:
+                test: null
+                small:
+                    - credit-g
+                    - vehicle
+            amlb_constraint:
+                - test
+    """
+    if "config_file" not in event or not event["config_file"].startswith("s3"):
+        raise KeyError("S3 path of config file is required.")
+
+    config_file_path = download_config(s3_path=event["config_file"])
+    with open(config_file_path, "r") as f:
+        configs = yaml.safe_load(f)
+
+    del configs["cdk_context"]
+
+    batch_job_queue = os.environ.get("BATCH_JOB_QUEUE")
+    batch_job_definition = os.environ.get("BATCH_JOB_DEFINITION")
+
+    module_configs = configs["module_configs"].pop(configs["module"])
+    del configs["module_configs"]
+    common_configs = configs
+    common_configs["mode"] = "local"
+    metrics_bucket = common_configs["metrics_bucket"]
+    common_configs = {
+        key: [value] if not isinstance(value, list) else value
+        for key, value in common_configs.items()
+    }
+
+    if common_configs["module"][0] == "tabular":
+        amlb_benchmarks = module_configs.pop("amlb_benchmark", [])
+        amlb_tasks = module_configs.pop("amlb_task", {})
+
+    # Generate and print all combinations
+    for common_combination in product(
+        *[common_configs[key] for key in common_configs.keys()]
+    ):
+        for module_combination in product(
+            *[module_configs[key] for key in module_configs.keys()]
+        ):
+            if common_configs["module"][0] == "tabular":
+                for amlb_benchmark in amlb_benchmarks:
+                    amlb_task_values = amlb_tasks[amlb_benchmark]
+                    if amlb_task_values is None:
+                        amlb_task_values = [None]
+                    for amlb_task in amlb_task_values:
+                        combination = (
+                            common_combination
+                            + module_combination
+                            + (amlb_benchmark, amlb_task)
+                        )
```

Comment:
These four layers of for loop looks nasty, can we simplify by moving some layers into functions example?